Aliucord/src/main/java/com/aliucord/coreplugins/NewPins.kt

```diff
@@ -0,0 +1,300 @@
+                    result.canDelete,
+                    result.canAddReactions,
+                    canPin,
+                    result.canMarkUnread
+                )
+            )
+        }
+    }
+
+    // Patches the permissions editor to show new permission
+    private fun patchPermissionsEditor() {
+        val pinMessagesCheckboxViewId = View.generateViewId()
+
+        // Patches the editor to add our new pin messages checkbox and change the information
+        // regarding the manage messages permission
+        patcher.before<WidgetChannelSettingsEditPermissions>(
+            "onViewBound",
+            View::class.java,
+        ) { (_, view: View) ->
+            val layoutId = Utils.getResId("channel_permissions_text_container", "id")
+            val layout: LinearLayout = view.findViewById(layoutId)
+            val manageMessagesCheckboxId = Utils.getResId("channel_permission_text_manage_messages", "id")
+            val manageMessagesCheckbox: TernaryCheckBox = view.findViewById(manageMessagesCheckboxId)
+            val index = layout.indexOfChild(manageMessagesCheckbox) + 1
+
+            // TODO: Should be changed after 2026/01/12, or when official client does
+            manageMessagesCheckbox.setLabel("Manage Messages ⚠")
+            manageMessagesCheckbox.setSubtext(
+                "Members with this permission can delete messages by other members or pin any message.*\n\n" +
+                "* Pinning messages now has a separate permission. This setting's behaviour will change soon."
+            )
+
+            TernaryCheckBox(view.context, null).addTo(layout, index) {
+                id = pinMessagesCheckboxViewId
+                layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
+
+                this.k.e.run { /* label.run */
+                    text = "Pin Messages"
+                    visibility = View.VISIBLE
+                }
+
+                this.k.f.run { /* subtext.run */
+                    text = "Allows members to pin or unpin any message."
+                    visibility = View.VISIBLE
+                }
+            }
+        }
+
+        // Patches logic to map our new permission checkbox to its value
+        patcher.before<WidgetChannelSettingsEditPermissions.Companion>(
+            "getPermission",
+            Int::class.javaPrimitiveType!!,
+        ) { (param, resId: Int) ->
+            if (resId == pinMessagesCheckboxViewId) {
+                param.result = PIN_MESSAGES_PERMISSION
+            }
+        }
+
+        // Patches logic that handles configuring initial state of checkboxes
+        patcher.after<`WidgetChannelSettingsEditPermissions$permissionCheckboxes$2`>("invoke") { param ->
+            val widget = this.`this$0`
+            val view = widget.view
+            if (view == null) {
+                logger.error(IllegalStateException("View was not initialised"))
+                return@after
+            }
```

Comment:
```kt
            val view = this.`this$0`.view
            if (view == null) {
                logger.error(IllegalStateException("View was not initialized"))
                return@after
            }
```

Reply:
Since `this$0` is unclear then it's better to leave this as-is